airflow/api/common/experimental/get_dag_run_state.py

```diff
@@ -19,9 +19,12 @@
 from datetime import datetime
 from typing import Dict
 
+from deprecated import deprecated
+
 from airflow.api.common.experimental import check_and_get_dag, check_and_get_dagrun
 
 
+@deprecated(reason="Use DagRun().get_state() instead", version="2.2.3")
```

Comment:
This is still used in airflow/www/api/experimental/endpoints.py -- if we are deprecating it we will need to change those references too.

Reply:
The experimental API is also deprecated: See https://github.com/apache/airflow/blob/b20e6d3f060bc385e350433070d5707ae6d6d0b0/airflow/www/api/experimental/endpoints.py#L56-L60.

My thinking is that if someone is using it externally which may be possible, then we should warn